osfoffline/polling_osf_manager/polling_events.py

```diff
@@ -21,117 +21,150 @@ def run(self):
 
 class CreateFolder(PollingEvent):
     def __init__(self, path):
-        super().__init__(path)
-        self.path = ProperPath(path, is_dir=True)
-        assert self.path
+        super().__init__()
+        self.path = path
 
     @asyncio.coroutine
     def run(self):
         # create local node folder on filesystem
-        if not os.path.exists(self.path.full_path):
-            AlertHandler.info(self.path.name, AlertHandler.DOWNLOAD)
-            os.makedirs(self.path.full_path)
+        try:
+            folder_to_create = ProperPath(self.path, is_dir=True)
+        except Exception:
+            # TODO: Narrow down this exception and do client side warnings
+            logging.exception('Exception caught: Invalid target path for folder.')
+            return
+        if not os.path.exists(folder_to_create.full_path):
+            AlertHandler.info(folder_to_create.name, AlertHandler.DOWNLOAD)
+            os.makedirs(folder_to_create.full_path)
 
 
 class CreateFile(PollingEvent):
     def __init__(self, path, download_url, osf_query):
-        super().__init__(path)
-        self.path = ProperPath(path, is_dir=False)
+        super().__init__()
+        self.path = path
         self.osf_query = osf_query
         self.download_url = download_url
-        assert self.path
-        assert isinstance(self.osf_query, OSFQuery)
-        assert isinstance(self.download_url, str)
 
     @asyncio.coroutine
     def run(self):
-        AlertHandler.info(self.path.name, AlertHandler.DOWNLOAD)
-        yield from _download_file(self.path, self.download_url, self.osf_query)
+        try:
+            new_file_path = ProperPath(self.path, is_dir=False)
+        except Exception:
+            # TODO: Narrow down this exception and do client side warnings
+            logging.exception('Exception caught: Invalid target path for new file.')
+            return
+        AlertHandler.info(new_file_path.name, AlertHandler.DOWNLOAD)
+        yield from _download_file(new_file_path, self.download_url, self.osf_query)
```

Comment:
As far as I can tell, Download file raises does handle OS errors (if it wouldn't be allowed to create the file locally). This can be tested by trying to sync a project anew from the OSF with a bad filename.